src/device-registry/utils/generate-filter.js

```diff
@@ -62,65 +63,74 @@ const generateDeviceFilter = (
   channel,
   location,
   siteName,
-  mapAddress
+  mapAddress,
+  primary,
+  active
 ) => {
-  if (tenant && name && !channel && !location && !siteName && !mapAddress) {
+
+  if(!tenant){
```

Comment:
Thanks @NoahNsimbe 
We shall need to review this logic here.
In case it is only the tenant provided, then we need to generate an empty filter.
Hence something along the lines of:
```
if (tenant){
return {}
}
```
Or better still, employ multiple else if statements and end with the default one which returns an empty object. More like:
```
if (name){
filter['name'] = { $regex: regexExpression, $options: "i" };
}
else if (channel){
filter['channelID'] = channel;
}
else if (location){
filter['locationID'] = location;
}
.....etc
else {
return {}
}
```

Actually, tenant is not required in this use case for generating filters.

We also need to review all possible filter combinations in the process. Example: what if one provides both the location and  the channel in the query parameters? What kind of filter shall we have then? I assume the filter there would be: 
```
{
channelID: channel,
locationID: location
}
```
This will then lead us to review the way the assignments are being done throughout this filtration process. 
In summary, we need to review all possible query combinations and test them out accordingly.

Reply:
@Baalmart, I have used `if` statements, meaning that if somebody provides for than one query parameter, they are all captured, for example checkout the following in the code. With approach, I believe a query filter is captured as long as it has been provided because execution continues till the return statement. The logic is that if a filter is provided, it is added to the `filter` variable and execution continues to check for other filters, if the filter is not provided, still execution continues and checks for other filters
```
 if(channel)
{    filter['channelID'] = channel;  }  
if(location)
{    filter['locationID'] = location;  }
```

Even if they provide only the tenant, `{}` is returned. Checkout `let filter = {};` in the code that is returned at the end.

Reply:
Makes sense @NoahNsimbe , thanks for the clarification.